### PR TITLE
KTOR-4105 Fixed ContentType.parser crash in Darwin and Browserjs in some case

### PR DESCRIPTION
**Pull request by roths**
…ome case

**Subsystem**
ktor-client, related modules : ktor-client-core.jsMain, ktor-client-darwin.darwinMain

**Motivation**
https://youtrack.jetbrains.com/issue/KTOR-4105

**Solution**
split header value with ',' before apend them into HttpHeadersBuilder

### CODE REVIEW

**Review by rsinukov — CHANGES_REQUESTED** [review]
Thanks for you PR.
Can you please add a test for it inside `ktor-client-tests` module?

**Comment by roths** [comment]
> Thanks for you PR. Can you please add a test for it inside `ktor-client-tests` module?

Already added

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/ContentTest.kt [hunk 1]
```diff
@@ -291,6 +291,17 @@ class ContentTest : ClientLoader(5 * 60) {
         }
     }
 
+    @Test
+    fun testDuplicatedContentType() = clientTests(listOf("CIO", "Android", "Apache", "Curl", "Java", "Jetty", "Okhttp")) {
```
Comment (rsinukov) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/ContentTest.kt:295]: Is there a reason to ignore all these engines?

File: ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinResponseReader.kt [hunk 2]
```diff
@@ -36,7 +36,11 @@ internal class DarwinResponseReader(
 
         val status = HttpStatusCode.fromValue(response.statusCode.toInt())
         val headers = buildHeaders {
-            headersDict.mapKeys { (key, value) -> append(key, value) }
+            headersDict.mapKeys { entry ->
+                entry.value.split(",").forEach{ valueItem ->
```
Comment (rsinukov) [inline ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinResponseReader.kt:40]: please fix codestyle here

**Review by rsinukov — CHANGES_REQUESTED** [review]
Thanks! Couple of comments

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/ContentTest.kt (hunk 1 above)
Comment (roths) [reply ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/ContentTest.kt:295]: because only browserjs and Darwin engine have this bug

Comment (rsinukov) [reply ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/ContentTest.kt:295]: But it's still good to verify this behaviour for other engines.

**Comment by rsinukov** [comment]
@e5l is it obsolete now?

**Comment by e5l** [comment]
Yep, it should be. Moreover, our parser can handle multiple content-type in one header line

**Comment by rsinukov** [comment]
Closing this PR as obsolete.

**Pull request closed without merging**